lambdas/archive-vcs/src/main/java/uk/gov/di/ipv/core/archivevcs/ArchiveVcsHandler.java

```diff
@@ -2,42 +2,100 @@
+import uk.gov.di.ipv.core.library.persistence.DataStore;
+import uk.gov.di.ipv.core.library.persistence.item.VcStoreItem;
 import uk.gov.di.ipv.core.library.service.ConfigService;
 import uk.gov.di.ipv.core.library.verifiablecredential.service.VerifiableCredentialService;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.List;
 
 @ExcludeFromGeneratedCoverageReport
 @SuppressWarnings("unused") // Temporarily disable to pass sonarqube
 public class ArchiveVcsHandler implements RequestStreamHandler {
     private static final Logger LOGGER = LogManager.getLogger();
     private final ConfigService configService;
     private final VerifiableCredentialService verifiableCredentialService;
+    private final DataStore<VcStoreItem> dataStore;
 
     @SuppressWarnings("unused") // Used by AWS
     public ArchiveVcsHandler(
-            ConfigService configService, VerifiableCredentialService verifiableCredentialService) {
+            ConfigService configService,
+            VerifiableCredentialService verifiableCredentialService,
+            DataStore<VcStoreItem> dataStore) {
         this.configService = configService;
         this.verifiableCredentialService = verifiableCredentialService;
+        this.dataStore = dataStore;
     }
 
     @SuppressWarnings("unused") // Used through dependency injection
     public ArchiveVcsHandler() {
         this.configService = new ConfigService();
         this.verifiableCredentialService = new VerifiableCredentialService(configService);
+        boolean isRunningLocally = this.configService.isRunningLocally();
+        this.dataStore =
+                new DataStore<>(
+                        this.configService.getEnvironmentVariable(
+                                EnvironmentVariable.ARCHIVED_USER_CREDENTIALS_TABLE_NAME),
+                        VcStoreItem.class,
+                        DataStore.getClient(isRunningLocally),
+                        isRunningLocally,
+                        configService);
     }
 
     @Override
     @Tracing
     @Logging(clearState = true)
     public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) {
         LogHelper.attachComponentIdToLogs(configService);
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            List<String> userIds = mapper.readValue(inputStream, RevokeRequest.class).getUserIds();
+            LOGGER.info("Revoking {} VCs", userIds.size());
+
+            List<List<String>> batchedRequest = ListHelper.getBatches(userIds, 100);
+            for (int i = 0; i < batchedRequest.size(); i++) {
+                LOGGER.info("Processing batch {} of {}", i, batchedRequest.size());
+                revoke(batchedRequest.get(i));
+            }
+            LOGGER.info("Completed revoking {} VCs", userIds.size());
+        } catch (IOException e) {
+            throw new FailedVcRevokeException("Failed to map the revoke request", e);
```

Comment:
(there's probably limited value in rethrowing here since it's never caught - it'll just fail the lambda anyway, but doesn't matter very much)